qt/scientific_interfaces/Indirect/IndirectFitAnalysisTab.cpp

```diff
@@ -460,267 +860,566 @@ void IndirectFitAnalysisTab::updatePlot(
+  algorithm->setProperty("WorkspaceIndex",
+                         m_fitPropertyBrowser->workspaceIndex());
+  return algorithm;
+}
+
+/**
+ * @return The current sequential fit algorithm for this indirect fit analysis
+ *         tab.
+ */
+IAlgorithm_sptr IndirectFitAnalysisTab::sequentialFitAlgorithm() const {
+  return singleFitAlgorithm();
+}
+
+/**
+ * Executes the single fit algorithm defined in this indirect fit analysis tab.
+ */
+void IndirectFitAnalysisTab::executeSingleFit() {
+  m_outputFitName = createSingleFitOutputName();
+  runFitAlgorithm(singleFitAlgorithm());
+}
+
+/**
+ * Executes the sequential fit algorithm defined in this indirect fit analysis
+ * tab.
+ */
+void IndirectFitAnalysisTab::executeSequentialFit() {
+  m_outputFitName = createSequentialFitOutputName();
+  runFitAlgorithm(sequentialFitAlgorithm());
+}
+
+/**
+ * @return  The fit function defined in this indirect fit analysis tab.
+ */
+IFunction_sptr IndirectFitAnalysisTab::fitFunction() const {
+  if (!emptyModel())
+    return m_fitPropertyBrowser->getFittingFunction();
+  else
+    return nullptr;
+}
+
+/**
+ * @param function  The function in the fit property browser.
+ * @return          A map from the name of a function in the fit property
+ *                  browser, to the name of a function in the selected model.
+ */
+QHash<QString, QString>
+    IndirectFitAnalysisTab::functionNameChanges(IFunction_sptr) const {
+  return QHash<QString, QString>();
+}
+
+/**
+ * @return  The workspace containing the data to be fit.
+ */
+MatrixWorkspace_sptr IndirectFitAnalysisTab::fitWorkspace() const {
+  return boost::dynamic_pointer_cast<MatrixWorkspace>(
+      m_fitPropertyBrowser->getWorkspace());
+}
+
+/**
+ * Sets the MaxIterations property of the specified algorithm, to the specified
+ * integer value.
+ *
+ * @param fitAlgorithm  The fit algorithm whose MaxIterations property to set.
+ * @param maxIterations The value to set.
+ */
+void IndirectFitAnalysisTab::setMaxIterations(IAlgorithm_sptr fitAlgorithm,
+                                              int maxIterations) const {
+  setAlgorithmProperty(fitAlgorithm, "MaxIterations", maxIterations);
+}
+
 /*
  * Runs the specified fit algorithm and calls the algorithmComplete
  * method of this fit analysis tab once completed.
  *
  * @param fitAlgorithm      The fit algorithm to run.
  */
-void IndirectFitAnalysisTab::runFitAlgorithm(
-    Mantid::API::IAlgorithm_sptr fitAlgorithm) {
+void IndirectFitAnalysisTab::runFitAlgorithm(IAlgorithm_sptr fitAlgorithm) {
+  disconnect(m_fitPropertyBrowser, SIGNAL(parameterChanged(const IFunction *)),
+             this, SLOT(updateGuessPlots()));
+  m_functionNameChanges = functionNameChanges(model());
+  auto function = fitFunction();
+  if (!m_functionNameChanges.isEmpty())
+    function = updateFunctionTies(function, m_functionNameChanges);
+  function->applyTies();
+
+  setAlgorithmProperty(fitAlgorithm, "InputWorkspace", fitWorkspace());
```

Comment:
Why cannot you just `fitAlgorithm->setProperty("InputWorkspace", fitWorkspace());`?